components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthnMissingClaimHandler.java

```diff
@@ -171,10 +177,36 @@ protected PostAuthnHandlerFlowStatus handlePostAuthenticationForMissingClaimsReq
                     }
                 }
 
+                List<LocalClaim> localClaims = getClaimMetadataManagementService().getLocalClaims(context.getTenantDomain());
+                Map<String, String> displayName = new HashMap<>();
+
+                for (Map.Entry<String, String> entry : missingClaimMap.entrySet()) {
+                    for (LocalClaim localClaim : localClaims){
+                        if (entry.getValue().equalsIgnoreCase(localClaim.getClaimURI())) {
+                            displayName.put(entry.getValue(), localClaim.getClaimProperties().get(DISPLAY_NAME_PROPERTY));
+                            break;
+                        }
+                    }
+                }
+
+                StringBuilder displayNameString = new StringBuilder();
+                StringBuilder displayNameValuesString = new StringBuilder();
+
+                for (Map.Entry<String, String> claimDisplay : displayName.entrySet()) {
+                    displayNameString.append(claimDisplay.getKey());
+                    displayNameValuesString.append(claimDisplay.getValue());
+                    displayNameString.append(",");
+                    displayNameValuesString.append(",");
```

Comment:
Wont this leave a "," at the end of the string. Using StringJoiner can avoid this